### ISSUE TITLE
Add pgVector in the building instructions for the PostgreSQL Docker image

### ISSUE THREAD

**Issue by Harvester62** [langflow-ai/langflow#3718]
### Feature Request

Include the pgVector component with the Docker build instructions. This would provide the use with a fully functional PostgreSQL Vector DB, ready to be used inside LangFlow.

### Motivation

I am not a programmer, neither I do have proper knowledge of SQL, but I liked to play with some RAG ideas and LangFlow seems perfect. 
So, after installing the Docker version for development of LangFlow, I noticed that the PostgreSQL server is missing the pgVector component, or at least that is what I understood from the error messages. 
Perhaps, it would be useful if the pgVector could be included in the Docker container, so having the user to just activate it on the SQL database. Anyway, I might be wrong, so in that case please forgive me.

### Your Contribution

After looking into the repository and searching around, with the help of AI (of course!), I found that the Docker instructions for the PostgreSQL server are defined inside the file \docker\cdk.Dockerfile (hope it's correct), and these might be the instructions to include pgVector:

```
FROM --platform=linux/amd64 python:3.10-slim

WORKDIR /app

# Install Poetry and build dependencies
RUN apt-get update && apt-get install -y \
    gcc \
    g++ \
    curl \
    build-essential \
    git \
    postgresql-server-dev-all \
    && rm -rf /var/lib/apt/lists/*

# Install Poetry
RUN curl -sSL https://install.python-poetry.org | python3 -

# Add Poetry to PATH
ENV PATH="${PATH}:/root/.local/bin"

# Copy the pyproject.toml and poetry.lock files
COPY poetry.lock pyproject.toml ./

# Copy the rest of the application codes
COPY ./ ./

# Install dependencies
RUN poetry config virtualenvs.create false && poetry install --no-interaction --no-ansi

# Install pgvector extension
RUN git clone https://github.com/pgvector/pgvector.git /tmp/pgvector && \
    cd /tmp/pgvector && \
    make && \
    make install && \
    rm -rf /tmp/pgvector

# Install additional dependencies
RUN poetry add botocore
RUN poetry add pymysql

# Command to run your application
CMD ["sh", "./container-cmd-cdk.sh"]
```

**Comment by Harvester62** [comment]
I know, it doesn't work, but I also advised you that I am not an educated programmer. I shouldn't have added the code, I apologize. Let's consider just the feature request main content and forget the rest. This ticket can be removed if useless.

**Comment by Harvester62** [comment]
Finally, I found the solution to my problem, which was to simply to change the image in the \docker_example\docker-compose.yml with one that has already pgVector installed (eg. pgvector/pgvector:pg16), rebuild the container, then activate the extension with 

`docker exec -it [docker container name] psql -U langflow -c "CREATE EXTENSION vector"`

The PostgreSQL Server Connection String I used in Langflow is `postgresql://langflow:langflow@postgres:5432/langflow`

After that I have been able to use pgVector inside Langflow.  Having solved it I am closing this bug report.
